### PR TITLE
Add `encoding` selector option (base64 or binary)

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

Add `encoding` option to select the encoding of Avro payload between `base64` (default) and `binary`

## What does this PR do?
Adds `encoding`  set option to choose the encoding of Avro payload.
By default it's set to `base64`.

## Why is it important/What is the impact to the user?

Let the user to decide  whether to use `binary` or `base64` encoding.

## Checklist

- [x] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
- ~~[ ] I have made corresponding change to the default configuration files (and/or docker env variables)~~
- [x] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [x] run this gem in Logstash

## How to test this PR locally

- clone this branch and configure `Gemfile`
- execute the pipeline:
```
input {
  generator {
    message => '{"id": "my account", "amount": 5.0}'
    count => 1 
    codec => json
  }
}

output {
  stdout {
    codec => avro {
      schema_uri => "/tmp/avro_schema_payment.asvc"
      base64_encoding => false
    }
  }
}
```
- Avro schema to save in  `/tmp/avro_schema_payment.asvc`
```
{"namespace": "io.confluent.examples.clients.basicavro",
 "type": "record",
 "name": "Payment",
 "fields": [
     {"name": "id", "type": "string"},
     {"name": "amount", "type": "double"}
 ]
}
```

## Related issues

- Closes #30

## Use cases
As user I want to receive the binary Avro payload without base64 encoding.

### CODE REVIEW

**Comment by andsel** [comment]
I would ask if @karenzone could give a quick check to the documentation change

File: docs/index.asciidoc [hunk 1]
```diff
@@ -99,6 +100,16 @@ output {
 
 Controls this plugin's compatibility with the {ecs-ref}[Elastic Common Schema (ECS)].
 
+[id="plugins-{type}s-{plugin}-base64_encoding"]
+===== `base64_encoding`
```
Comment (yaauie) [inline docs/index.asciidoc:104]: I would prefer we avoid adding boolean options tied to the current implementation detail of there being two options, instead opting for a name that is descriptive of what the option controls and a value that is descriptive of what is being done. This allows us to leave room for additional options in the future, such as alternate encodings that may be introduced upstream.

In this case, something like:

`encoding` with values `base64` and `binary`.

Comment (karenzone) [reply docs/index.asciidoc:104]: +1 on previous comment.  If you take this suggestion, the doc will need to be reworded. Please ping me and I'll review/retest doc then.   :-)

Comment (andsel) [reply docs/index.asciidoc]: Thanks for suggesting it. I've applied the hint, so it's ready for a review :-)

File: docs/index.asciidoc [hunk 2]
```diff
@@ -99,6 +100,16 @@ output {
...
+[id="plugins-{type}s-{plugin}-encoding"]
+===== `encoding`
+
+* Value can be any of: `binary`, `base64`
+* Default value is `base64`
+
+Select which encoding to use to represent Avro's payload.
+By default, it uses `base64`, however could be customized with `binary` to use the plain binary bytes.
+With `base64` encoding the raw binary bytes are converted to a `base64` encoded string.
+
```
Comment (karenzone) [inline docs/index.asciidoc:112]:
```suggestion
Set encoding for Avro's payload. 
Use `base64` (default) encoding to convert the raw binary bytes to a `base64` encoded string.

Set this option to `binary` to use the plain binary bytes.
```

**Review by karenzone — COMMENTED** [review]
I left a suggestion inline to streamline and simplify the wording.   Otherwise, LGTM

File: docs/index.asciidoc (hunk 2 above)
Comment (andsel) [reply docs/index.asciidoc]: Thank's @karenzone accepted your wonderful suggestion! :-)

File: lib/logstash/codecs/avro.rb [hunk 3]
```diff
@@ -59,6 +59,14 @@ class LogStash::Codecs::Avro < LogStash::Codecs::Base
 
   include LogStash::PluginMixins::EventSupport::EventFactoryAdapter
 
+  BINARY_ENCODING = "binary"
+  BASE64_ENCODING = "base64"
```
Comment (yaauie) [inline lib/logstash/codecs/avro.rb:63]: nitpick: strings are mutable, so if we are constantizing them we should also be freezing them.

```suggestion
  BINARY_ENCODING = "binary".freeze
  BASE64_ENCODING = "base64".freeze
```

File: docs/index.asciidoc [hunk 4]
```diff
@@ -99,6 +100,16 @@ output {
...
+[id="plugins-{type}s-{plugin}-encoding"]
+===== `encoding`
+
+* Value can be any of: `binary`, `base64`
+* Default value is `base64`
+
+Set encoding for Avro's payload. 
+Use `base64` (default) encoding to convert the raw binary bytes to a `base64` encoded string.
+
+Set this option to `binary` to use the plain binary bytes.
```
Comment (yaauie) [inline docs/index.asciidoc:112]:
```suggestion
Use `base64` (default) to indicate that this codec sends or expects to receive base64-encoded bytes.

Set this option to `binary` to indicate that this codec sends or expects to receive binary Avro data.

```

**Review by yaauie — APPROVED** [review]
In general, LGTM. I've left two optional nitpick suggestions.

**Comment by andsel** [comment]
Published with https://rubygems.org/gems/logstash-codec-avro/versions/3.4.0-java
